deel/lip/activations.py

```diff
@@ -126,20 +126,22 @@ def build(self, input_shape):
             self.n = input_shape[self.channel_axis]
         if (input_shape[self.channel_axis] % self.n) != 0:
             raise RuntimeError("self.n has to be a divisor of the number of channels")
+        self.flat_shape = [-1, tf.reduce_prod(input_shape[1:]).numpy() // self.n, self.n]
+        self.out_shape = tuple((-1,)) + tuple(input_shape.as_list()[1:])
 
     def _compute_lip_coef(self, input_shape=None):
         return 1.0
 
     def call(self, x, **kwargs):
-        fv = tf.reshape(x, [-1, self.n])
+        fv = tf.reshape(x, self.flat_shape)
         if self.n == 2:
             b, c = tf.split(fv, 2, 1)
-            newv = tf.concat([tf.minimum(b, c), tf.maximum(b, c)], axis=1)
-            newv = tf.reshape(newv, tf.shape(x))
-            return newv
+            newv = tf.concat([tf.minimum(b, c), tf.maximum(b, c)], axis=-1)
+            newv = tf.reshape(newv, self.out_shape)
+            return newv * self._get_coef()
```

Comment:
Out of curiosity, why did this `_get_coef()` appear here?

Reply:
this is because `GroupSort` inherits the `LipschitzLayer` class and hence allow the setting of a lipschitz constant. Since it has no weights, it is done by scaling the output.
However the pertinence of this in the case of activations is discutable, and it could be relevant to remove it in further releases.